hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/transport/server/ratis/ContainerStateMachine.java

```diff
@@ -249,6 +249,9 @@ public long takeSnapshot() throws IOException {
       LOG.info("Taking a snapshot to file {}", snapshotFile);
       try (FileOutputStream fos = new FileOutputStream(snapshotFile)) {
         persistContainerSet(fos);
+        fos.flush();
+        // make sure the snapshot file is synced
```

Comment:
why do we need to make sure to flush to disk. 
Asking this because in OM also we take a snapshot with lastappliedIndex value. So, trying to understand do we need to do it over there also.
 Want to understand why do we need to do this/ has this caused any issues?

Reply:
We need to persist the state on disk to ensure we don't read an incorrect replica state on restart. We have come across an issue where, unhealthy state of a replica was not persisted on disk and on restart , it was detected as an open replica which may end up taking more IO on restart causing inconsistencies.